turbo/snapshotsync/snapshotsync.go

```diff
@@ -83,6 +82,9 @@ func WaitForDownloader(ctx context.Context, logPrefix string, histV3 bool, capli
 		return nil
 	}
 
+	snapshots.Close()
```

Comment:
This line is unclear. Erigon has RPCDaemon (which enabled from start) - user can send requests - will they work? how they will work with "closed object"? are we sure calling methods on "closed object" is safe? do we have unit-tests for it? is it anti-pattern or not?

Reply:
This is necessary because otherwise the downloader can't modify files that may have changed during download.

I think that this does not occur for torrents because the mmap process does a replace in place via the mapping process.

The web downloader downloads into a temporary file and then renames it.  This fails if the snapshot is holding the file open.

Arguably in both cases the snapshots should be closed during the download operation - because the files are in an uncertain state and are not usable until the indexing process is complete.

We could make this more granular by passing the snapshots to the downloader then we could close the individual segments.

Reply:
I guess blockReader will see `sn.BlocksAvailable() = 0` until snapshots get indexed.